### PR TITLE
Add New Jersey total income variable and use it in exclusion formulas

### LINKED ISSUE #2839: New Jersey total income
The NJ income tax has a variable called total income (NJ-1040 form line 27) that is used extensively in the pension/retirement income exclusion calculations.  However, PolicyEngine-US does not have this variable.  Adding a `nj_total_income` Person variable would allow a major simplification of exclusion code and allow that code to be refactored (to eliminate code duplication and make model execution be slightly faster).

### PR DESCRIPTION
**Pull request by martinholmer**
Fixes #2839.  After these changes there are no NJ tax differences with the 2023-08-15 patched version of TAXSIM35 in the 300,000 integration test cases in the `p21`, `e21`, and `f21` samples described in issue #993.

### Summary
:memo::sparkles::wrench:

This pull request improves the accuracy and consistency of the New Jersey income tax parameters and variables, especially related to the retirement income exclusions. It updates the `percentage` parameters for the pension exclusion for different filing statuses, using correct terminology and brackets. It adds a new parameter file for the other exclusions from total income, and modifies the description of the subtractions parameter. It also changes the entity and label of the additions and subtractions variables, and adds new variables for the total income, the eligible pension income, and the other retirement income exclusion. It updates and adds test cases for the New Jersey income tax variables, and deletes some redundant or obsolete test files. It also adds an empty file for the changelog entries.

> _`New Jersey` tax code_
> _Adjusting income and tests_
> _Autumn leaves fall fast_

### Walkthrough
*  Add a new YAML file to store the changelog entries for the pull request ([link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1))
*  Add a new YAML file to define the parameters for the New Jersey exclusions from total income ([link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-3f5406c67027a8f49dc79ec98a2e3d40a17b4a22debf9059a9e8327c77605590R1-R9))
*  Modify the descriptions, labels, and brackets of the percentage parameters for the New Jersey pension retirement exclusion, for each filing status, to use the term fraction instead of percentage, and to follow the state tax instructions ([link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-580e08f9c84a6f1628603a9eda7587d43ef57633d0a7c4299e8440a1ffe8637bL1-R1), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-580e08f9c84a6f1628603a9eda7587d43ef57633d0a7c4299e8440a1ffe8637bL7-R7), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-580e08f9c84a6f1628603a9eda7587d43ef57633d0a7c4299e8440a1ffe8637bL13-R25), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-8c9f903baca342f601a6057e8948f0338faca220a6dc50522193e63368ff71bbL1-R1), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-8c9f903baca342f601a6057e8948f0338faca220a6dc50522193e63368ff71bbL7-R7), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-8c9f903baca342f601a6057e8948f0338faca220a6dc50522193e63368ff71bbL13-R25), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-c7460bc5249534fd5ce38c8c8a43cddf2445c96c00f34c9dd0559e8a49364ffdL1-R1), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-c7460bc5249534fd5ce38c8c8a43cddf2445c96c00f34c9dd0559e8a49364ffdL7-R7), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-c7460bc5249534fd5ce38c8c8a43cddf2445c96c00f34c9dd0559e8a49364ffdL13-R25), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-e7d963d3f3b5844af51050349a926f127cf55a5922d79b7ae9aecb3e337bcef6L1-R1), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-e7d963d3f3b5844af51050349a926f127cf55a5922d79b7ae9aecb3e337bcef6L7-R7), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-e7d963d3f3b5844af51050349a926f127cf55a5922d79b7ae9aecb3e337bcef6L13-R25), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-6d3a7bed692a36156168be3b38f29b71f2084a95b0912462099ff0f1be244870L1-R1), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-6d3a7bed692a36156168be3b38f29b71f2084a95b0912462099ff0f1be244870L7-R7), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-6d3a7bed692a36156168be3b38f29b71f2084a95b0912462099ff0f1be244870L13-R25))
*  Modify the description of the subtractions parameter for the New Jersey adjusted gross income, to use the term federal instead of New Jersey, and reorder the metadata and values fields for consistency ([link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-a4d1bf672cac30e1393f0bc55691546e5579a75ee16e5b6b2a705f28d3b37440L1-R7))
*  Add a new Python file to define the variable for the New Jersey total income, which is the sum of the federal adjusted gross income and the additions and subtractions for each person in the tax unit ([link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-a479a73b46261ca52c5ede77a1c23f288bf5e8bc1b6cf27571a7e979e7e37dd4R1-R17))
*  Modify the definition and formula of the New Jersey AGI additions, to use the person entity instead of the tax unit entity, and to change the label and documentation to reflect that the additions are for the total income, not the adjusted gross income ([link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-05ce4c67cf6cc445018913e61273369818613a55fe1d2dcc9560f2fa8dae0409L6-R9), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-c617bce4230bb5eebd7c4054891c2358656439cb7c3f614b3c72823547597e97L14-R17))
*  Modify the definition of the New Jersey AGI subtractions, to use the person entity instead of the tax unit entity, and to change the label and documentation to reflect that the subtractions are for the total income, not the adjusted gross income ([link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-850dff8577d690ba4e5ad339bdfb305ffffdb70d289e88844e6e6db9d3e465acL6-R9))
*  Modify the formula for the New Jersey adjusted gross income, to use the total income and exclusions parameters instead of the federal adjusted gross income and additions and subtractions parameters ([link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-727688adc7b665e11e6de4143a474f3af7b7138888a322b9ceb1e53df339dd3dL1-R11))
*  Add a new Python file to define the variable for the New Jersey eligible pension income, which is the pension income that can be excluded from the total income for each person in the tax unit who meets the age, blindness, disability, and filing status criteria ([link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-918b998914048ddb08e06cad5a9a6cbc97b99bfcb22f81320d585859cd9607dbR1-R30))
*  Add a new Python file to define the variable for the New Jersey other retirement income exclusion, which is the amount of other retirement income that can be excluded from the total income for the tax unit, based on the unused portion of the pension exclusion, the total income, the filing status, and the earned income ([link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-08593804387f51e7fbf7c3e6586f8aa548839f5eab110e8c734ee03c6c5c5ac1R1-R53))
*  Add a new YAML file to define a test case for the New Jersey other retirement income exclusion, based on an example from the state tax instructions ([link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-ec9bbd2ff23e704241205e1e05448fb4f1d3aa0cb62c6f3388e903fe8e31205eL1-R34))
*  Add numbers to the names of the test cases for the New Jersey pension retirement exclusion, for consistency with the other test cases, and modify some of the inputs to reflect the examples from the state tax instructions ([link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-1d2e3b883d90ca126747a8a0241fcd229db1a4c7a925519bbca08633aa6d7785L1-R1), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-1d2e3b883d90ca126747a8a0241fcd229db1a4c7a925519bbca08633aa6d7785L32-R32), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-1d2e3b883d90ca126747a8a0241fcd229db1a4c7a925519bbca08633aa6d7785L63-R63), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-1d2e3b883d90ca126747a8a0241fcd229db1a4c7a925519bbca08633aa6d7785L99-R99), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-1d2e3b883d90ca126747a8a0241fcd229db1a4c7a925519bbca08633aa6d7785L130-R130), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-1d2e3b883d90ca126747a8a0241fcd229db1a4c7a925519bbca08633aa6d7785L162-R162), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-1d2e3b883d90ca126747a8a0241fcd229db1a4c7a925519bbca08633aa6d7785L194-R194), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-1d2e3b883d90ca126747a8a0241fcd229db1a4c7a925519bbca08633aa6d7785L220-R220), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-1d2e3b883d90ca126747a8a0241fcd229db1a4c7a925519bbca08633aa6d7785L246-R246), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-1d2e3b883d90ca126747a8a0241fcd229db1a4c7a925519bbca08633aa6d7785L278-R278), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-1d2e3b883d90ca126747a8a0241fcd229db1a4c7a925519bbca08633aa6d7785R288), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-1d2e3b883d90ca126747a8a0241fcd229db1a4c7a925519bbca08633aa6d7785R295), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-1d2e3b883d90ca126747a8a0241fcd229db1a4c7a925519bbca08633aa6d7785L301-L302), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-1d2e3b883d90ca126747a8a0241fcd229db1a4c7a925519bbca08633aa6d7785L310-R310), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-1d2e3b883d90ca126747a8a0241fcd229db1a4c7a925519bbca08633aa6d7785R320), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-1d2e3b883d90ca126747a8a0241fcd229db1a4c7a925519bbca08633aa6d7785R327), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-1d2e3b883d90ca126747a8a0241fcd229db1a4c7a925519bbca08633aa6d7785L333-L334), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-1d2e3b883d90ca126747a8a0241fcd229db1a4c7a925519bbca08633aa6d7785L342-R342), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-1d2e3b883d90ca126747a8a0241fcd229db1a4c7a925519bbca08633aa6d7785L352-R367), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-1d2e3b883d90ca126747a8a0241fcd229db1a4c7a925519bbca08633aa6d7785R377), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-1d2e3b883d90ca126747a8a0241fcd229db1a4c7a925519bbca08633aa6d7785L391-L392), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-1d2e3b883d90ca126747a8a0241fcd229db1a4c7a925519bbca08633aa6d7785L400-R392), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-1d2e3b883d90ca126747a8a0241fcd229db1a4c7a925519bbca08633aa6d7785L410-R418), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-1d2e3b883d90ca126747a8a0241fcd229db1a4c7a925519bbca08633aa6d7785L442-R439))
*  Add numbers to the names of the test cases for the New Jersey adjusted gross income, for consistency with the other test cases ([link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-727688adc7b665e11e6de4143a474f3af7b7138888a322b9ceb1e53df339dd3dL21-R21), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-727688adc7b665e11e6de4143a474f3af7b7138888a322b9ceb1e53df339dd3dL49-R49), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-727688adc7b665e11e6de4143a474f3af7b7138888a322b9ceb1e53df339dd3dL74-R80), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-727688adc7b665e11e6de4143a474f3af7b7138888a322b9ceb1e53df339dd3dL105-R106))
*  Delete three YAML files that were no longer needed for testing the New Jersey income tax calculations ([link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-63b7c2caa6f03add904e61b19c143ab4de03170655e5d5547403f98b8e3327ac), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-3bf4fa47e888de52ce18b27c66e3b192a9151674952c28c5eddbba952dc75963), [link](https://github.com/PolicyEngine/policyengine-us/pull/2840/files?diff=unified&w=0#diff-f3b8019c427d448b14a65031ac85b7498450a58524eb0b66dabbab0de83ce735))

### CODE REVIEW

File: changelog_entry.yaml
Comment (MaxGhenis) [inline changelog_entry.yaml:1]: populate

File: policyengine_us/variables/gov/states/nj/tax/income/exclusions/nj_other_retirement_income_exclusion.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        # follows Worksheet D
+        p = parameters(period).gov.states.nj.tax.income.exclusions.retirement
+
+        # determine age eligibility for head and spouse
+        person = tax_unit.members
+        is_head = person("is_tax_unit_head", period)
+        is_spouse = person("is_tax_unit_spouse", period)
+        age_eligible = person("age", period) >= p.age_threshold
+        age_eligible_unit = tax_unit.sum(
+            (age_eligible & is_head) | (age_eligible & is_spouse)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/nj/tax/income/exclusions/nj_other_retirement_income_exclusion.py:27]:
```suggestion
            age_eligible & (is_head | is_spouse)
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: changelog_entry.yaml
Comment (martinholmer) [reply changelog_entry.yaml:1]: Done

File: policyengine_us/variables/gov/states/nj/tax/income/exclusions/nj_other_retirement_income_exclusion.py (hunk 1 above)
Comment (martinholmer) [reply policyengine_us/variables/gov/states/nj/tax/income/exclusions/nj_other_retirement_income_exclusion.py:27]: Good suggestion.  Done in next push.

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**